### DISCUSSION TITLE
求助电信网络环境下，测试联通和移动线路速度。

### DISCUSSION THREAD

**Discussion by ZhiXuanWang** [XIU2/CloudflareSpeedTest#482] [提问] [answered]
上周，我开发了一个基于CloudflareSpeedTest定时脚本，用于优化 IP 地址选择，生成静态网页并发布到 GitHub。同时，我使用了 GitHub Actions 进行自动更新。为了方便大家使用，我还提供了 Top 接口，以及一个用于更新域名的功能。
然而，最近有人向我询问关于网络线路的问题。他发现，即使是使用相同的 IP 地址，联通和移动的网速和稳定性却存在显著差异。我思考了很长时间，但还没有找到一个满意的解决方案。因此，我希望能从大家那里得到一些关于这个问题的建议和解决方案。感谢。
相关链接：
https://ip.164746.xyz
https://github.com/zhixuanwang/cf-speed-dns
https://github.com/ZhiXuanWang/cf-speed-dns/issues/1

**Comment by XIU2** [comment, marked as answer]
这很正常吧，同一个 IP 不同运营商走的线路可能都不一样。
比如香港节点 IP 移动就能直连，而电信联通都要绕路。

如果想要足够精准，那么只能用户在自己的网络环境下测速。
其次的话，如果想要降低精准，尽量覆盖广的话，那就需要在国内三大运营商的环境下分别测速来获得三组 IP 来提供给对应运营商的人使用（包括其他那些做类似于你这种网站/服务的项目，也都是这样干的）。

**Reply by ZhiXuanWang** [reply]
感谢大佬回复。针对第二种方案成本较高，还是用户在自己的网络环境下进行测速吧，后续更新加入优选后的IP，提供ITDOG的TCPing测试结果供用户参考。不知道这方式有没有参考意义。

**Comment by yfdyh000** [comment]
同一个IP可能任播、BPG到不同的路由和地理位置。比如DNS服务的任播。
同一个运营商的不同地点、用户、时段，都可能有不同路由和QoS策略。比如专线或精品网的不绕路、优先保证。
